consumer/src/main/kotlin/streams/kafka/KafkaManualCommitEventConsumer.kt

```diff
@@ -3,15 +3,50 @@ package streams.kafka
 import org.apache.kafka.clients.consumer.OffsetAndMetadata
 import org.apache.kafka.common.TopicPartition
 import org.neo4j.logging.Log
+import streams.extensions.offsetAndMetadata
+import streams.extensions.topicPartition
 import streams.service.StreamsSinkEntity
-import streams.service.errors.ErrorService
+import java.time.Duration
 
 class KafkaManualCommitEventConsumer(config: KafkaSinkConfiguration,
-                                     log: Log): KafkaAutoCommitEventConsumer(config, log) {
+                                     private val log: Log,
+                                     topics: Set<String>): KafkaAutoCommitEventConsumer(config, log, topics) {
+
+    private val asyncCommit = config.streamsAsyncCommit
+
+    override fun stop() {
+        if (asyncCommit) {
+            /*
+             * While everything is fine, we use commitAsync.
+             * It is faster, and if one commit fails, the next commit will serve as a retry.
+             * But if we are closing, there is no "next commit". We call commitSync(),
+             * because it will retry until it succeeds or suffers unrecoverable failure.
+             */
+            consumer.commitSync()
+        }
+        super.stop()
+    }
 
     private fun commitData(commit: Boolean, topicMap: Map<TopicPartition, OffsetAndMetadata>) {
         if (commit) {
-            consumer.commitSync(topicMap)
+            if (asyncCommit) {
+                if (log.isDebugEnabled) {
+                    log.debug("Committing data in async")
```

Comment:
might want to remove these debug log statements for the final one.  It should be clear which happens by configuration, and this method gets called for every read, so this could be quite some logspam outside of testing